packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -200,6 +220,33 @@ export class WaferMap extends FoundationElement {
             DOM.queueUpdate(() => this.render());
         }
     }
+
+    private parseWaferDataToEventData(
```

Comment:
I think this data transformation should be the responsibility of the various handlers. Putting it in the wafer map means that the concepts like zoom handler and hover handler leak out to this level but it would be better to encapsulate those. The handlers could hopefully take a reference to the top level wafer map and use it to look up whatever they need and transform it to the form they need.

Reply:
I understand what you mean, and I did consider this approach during the refactor process, however the issue with this is that a lot of the required properties are set as private in the current implementation of the Wafermap, and such would not be accessible from within the `EventHandler `. I would definitely keep this as a potential future improvement idea, but I am not sure if it makes sense to do this kind of a change now, especially since these 4 elements (WaferMap, ZoomHandler, EventHandler and later the HoverHandler) are and will stay very much coupled ...

Reply:
I was worried that would be the case. The best workaround I can think of would be to make those wafer map properties `public` but tag them with `@internal` comments to indicate that they're not part of the public API. We do this for other components in quite a few places.

I still think this would be a step in the right direction to shrink the API surface of the handlers by encapsulating more. But I admit it might be messy or we might need to see how it works as we add more handlers so I'm ok deferring this.

Reply:
We can get back to this subject in subsequent PRs if it will be the case.